### PR TITLE
[Toggle]: Fix toggles not toggling

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/41493

**Note:** This should be uplifted to 1.72

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See issue

### CODE REVIEW

File: ui/webui/resources/cr_elements/cr_toggle/cr_toggle.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export class CrToggleElement extends CrLitElement {
       checked: {
         type: Boolean,
         reflect: true,
+        notify: true,
```
Comment (fallaciousreasoning) [inline ui/webui/resources/cr_elements/cr_toggle/cr_toggle.ts:35]: This was the root cause of the issue - it works a bit differently to the old polymer bindings

**Review by simonhong — APPROVED** [review]

File: ui/webui/resources/cr_elements/cr_toggle/cr_toggle.ts (hunk 1 above)
Comment (bsclifton) [reply ui/webui/resources/cr_elements/cr_toggle/cr_toggle.ts:35]: this is notifying the parent basically? per your comment below

**Review by bsclifton — COMMENTED** [review]
Fix works great for both cases in https://github.com/brave/brave-browser/issues/41493 🙂👍

**Review by bsclifton — APPROVED** [review]

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using
```
Brave | 1.73.20 Chromium: 130.0.6723.58 (Official Build) nightly (64-bit)
-- | --
Revision | 363777613b877e60655632bb3570327ce6baeafd
OS | Windows 11 Version 23H2 (Build 22631.4317)
```


1. Installed 1.73.20
2. launched Brave
3. created sync chain in the profile
4. created a second profile via hamburger menu
5. joined the sync chain
6. toggled Sync data on/off
7. added an extension via chrome webstore
8. opened `brave://settings/extensions` in a new tab
9. clicked Details
10. toggled on for `Allow in Private`

- Confirmed Sync data toggles worked as expected in both profiles
- Confirmed toggle worked for Private window and working as expected



https://github.com/user-attachments/assets/56f293ca-b3ee-4a18-9f6c-407833813c0c

**Pull request merged**